Instanceof redirect and synthetic field shadowing tutorial

BoogieMonster1O1 opened the pull request:
Credits to @biom4st3r for the instanceof redirect example

biom4st3r committed:
Example of instanceof mode redirect

biom4st3r committed:
Update redirect.md

BoogieMonster1O1 committed:
Merge branch 'master' into patch-1

BoogieMonster1O1 committed:
Prepare for a new tutorial

yes
i hath returned

BoogieMonster1O1 committed:
mouse bad

BoogieMonster1O1 committed:
mouse bad

BoogieMonster1O1 committed:
Merge pull request #1 from biom4st3r/patch-1

Example of instanceof mode redirect

BoogieMonster1O1 committed:
Create shadow-anonymous.md

BoogieMonster1O1 committed:
Update shadow-anonymous.md

dblsaiko commented on shadow-anonymous.md:
```diff
@@ -0,0 +1,43 @@
+### IntelliJ Idea
+- Open the class file
+
+![foo](https://i.imgur.com/hYWE1O3.png)
```
Could you put these inside the repo?

BoogieMonster1O1 replied:
Done, ive put them in `/assets`

dblsaiko commented on redirect.md:
````diff
@@ -36,4 +37,44 @@ Method modification:
+
+Example mixin:
+```java
+@Redirect(method = "target()Z", at = @At(value = "CONSTANT",args="classValue=net/example/Dummy",shift=Shift.AFTER,ordinal=0))
````
Please format the code consistently (spaces before and after "=", spaces after ",", braces on same line, consistent indentation)

dblsaiko reviewed (changes requested):
Interesting, didn't know you could redirect instanceof.

BoogieMonster1O1 committed:
Fix code style

BoogieMonster1O1 committed:
Assets

 Changes to be committed:
	new file:   assets/bytecode_1.png
	new file:   assets/bytecode_2.png
	new file:   assets/bytecode_3.png
	new file:   assets/bytecode_4.png
	modified:   shadow-anonymous.md
